library/psa_crypto_driver_wrappers.c

```diff
@@ -238,62 +238,56 @@ psa_status_t psa_driver_wrapper_verify_hash( psa_key_slot_t *slot,
 #endif /* PSA_CRYPTO_DRIVER_PRESENT */
 }
 
-#if defined(PSA_CRYPTO_ACCELERATOR_DRIVER_PRESENT)
-/** Calculate the size to allocate for buffering a key with given attributes.
- *
- * This function provides a way to get the expected size for storing a key with
- * the given attributes. This will be the size of the export representation for
- * cleartext keys, and a driver-defined size for keys stored by opaque drivers.
+/** Get the key buffer size for the key material of a generated key in the
+ *  case of an opaque driver without storage.
  *
- * \param[in] attributes        The key attribute structure of the key to store.
- * \param[out] expected_size    On success, a byte size large enough to contain
- *                              the declared key.
+ * \param[in] attributes  The key attributes.
+ * \param[out] key_buffer_size  Minimum buffer size to contain the key material
```

Comment:
If this is going to work for key import as well, I suggest making this an in/out variable for when the 'bits' attribute is set to 0 (typically on import) and autodetection of the key size is needed.

Specifically: when the bits attribute is set to 0, require that the value of `*key_buffer_size` is pre-set to the length of the data input for import, such that this function can return a worst-case key buffer size, and the import_key function call can set the bits attribute right (since import_key itself is not able to resize the key buffer).

Reply:
We can adapt this function to support the key import case as well. As this PR is only about key generation I would prefer to do this in a separate PR though.

Reply:
Well, this PR changes `psa_driver_wrapper_get_key_buffer_size` to be available externally. Doesn't it then make sense to thing about why it's being made external, which parts of the codebase are going to use it, and what those different parts require in terms of functionality of this entrypoint?

Reply:
I understand you point but:
. in this PR the documentation of `psa_driver_wrapper_get_key_buffer_size` clearly states that it is dedicated to key generation thus I don't think there is a need (would probably be better though) to go beyond that.
. given all PSA PRs that we currently merge, as this PR is quite large, it conflicts rapidly with development. As it is in a state suitable for merge now, I prefer to finalize it and do the change on `psa_driver_wrapper_get_key_buffer_size` in another PR. I will work on the change when this PR is merged.

Reply:
Guess I'm looking forward to your next PR then.